src/ray_agents/batch.py

```diff
@@ -0,0 +1,225 @@
+"""Batch tool wrapper for parallel execution of Ray tools."""
+
+import inspect
+from collections.abc import Callable
+from inspect import Signature
+from typing import Any
+
+import ray
+
+from ray_agents.adapters.core import RayTool, to_raytool
+
+
+def batch(
+    tool: Callable | RayTool,
+    description: str | None = None,
+) -> Callable:
+    """Create a batch version of a tool for parallel execution.
+
+    The batch tool allows LLMs to execute multiple instances of the same tool
+    in parallel with a single tool call, reducing round trips and improving
+    throughput.
+
+    Args:
+        tool: Any tool (decorated function, wrapped tool, or RayTool)
+        description: Optional custom description for the batch tool
+
+    Returns:
+        A new callable that accepts list inputs and executes in parallel.
+        Can be wrapped with RayToolWrapper for framework-specific format.
+
+    Example:
+        >>> from ray_agents import tool, batch
+        >>>
+        >>> @tool(desc="Fetch content from a URL")
+        >>> def fetch_url(url: str) -> str:
+        ...     return requests.get(url).text
```

Comment:
How does passing inputs work when there are multiple arguments to the tool? It may be useful to showcase or document such an example